Source/Request.swift

```diff
@@ -420,6 +438,8 @@ public class Request {
                 // An example of how this can happen is by calling cancel inside a response completion closure.
                 mutableState.responseSerializers.removeAll()
                 mutableState.responseSerializerCompletions.removeAll()
+
+                mutableState.state = .finished
```

Comment:
We only want to transition to `.finished` if we weren't cancelled, so this should probably use the state transition API.

Reply:
Agreed.

Reply:
Check added in 656f331d.